tools/rpcgen.py

```diff
@@ -103,6 +105,37 @@ class failure_probes;
       {%- endfor %}
     }
 
+    void setup_public_metrics() final {
+        namespace sm = ss::metrics;
+        auto component_label = sm::label("component");
+        auto service_label = sm::label("service");
+        auto method_label = sm::label("method");
+      {%- for method in methods %}
+        {
+            std::vector<ss::metrics::label_instance> labels{
+              service_label("{{service_name}}"),
+              method_label("{{method.name}}"),
+              component_label("internal")};
+                auto aggregate_labels
+                  = config::shard_local_cfg().aggregate_metrics()
+                      ? std::vector<sm::label>{sm::shard_label, method_label}
+                      : std::vector<sm::label>{};
```

Comment:
Indentation looks funny here, but we should always aggregate for `/public_metrics`.

Reply:
Done.